mubench.pipeline/tests/tasks/implementations/test_compile.py

```diff
@@ -294,7 +294,41 @@ def test_compile_with_gradle(self, shell_mock, copy_mock):
         shell_mock.return_value = ":printClasspath\n/path/dependency1.jar\n/path/dependency2.jar\n\nBUILD SUCCESSFUL"
 
         self.uut.process_project_version(self.project, self.version)
-        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b classpath.gradle",))
+        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b 'classpath.gradle'",))
+        assert_in(call("/path/dependency1.jar", self.dep_path), copy_mock.mock_calls)
+        assert_in(call("/path/dependency2.jar", self.dep_path), copy_mock.mock_calls)
+
+    @patch("tasks.implementations.compile.shutil.copy")
+    @patch("tasks.implementations.compile.Shell.exec")
+    def test_compile_with_gradle_project_dir_parameter(self, shell_mock, copy_mock):
+        self.version._YAML["build"]["commands"] = ["gradle build -p /testdir/"]
+        shell_mock.return_value = ":printClasspath\n/path/dependency1.jar\n/path/dependency2.jar\n\nBUILD SUCCESSFUL"
+
+        self.uut.process_project_version(self.project, self.version)
+        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b '/testdir/classpath.gradle'",))
+        assert_in(call("/path/dependency1.jar", self.dep_path), copy_mock.mock_calls)
+        assert_in(call("/path/dependency2.jar", self.dep_path), copy_mock.mock_calls)
+
+    @patch("tasks.implementations.compile.shutil.copy")
+    @patch("tasks.implementations.compile.Shell.exec")
+    def test_compile_with_gradle_project_dir_parameter_spaces(self, shell_mock, copy_mock):
+        self.version._YAML["build"]["commands"] = ["gradle build -p \"/test dir/\""]
+        shell_mock.return_value = ":printClasspath\n/path/dependency1.jar\n/path/dependency2.jar\n\nBUILD SUCCESSFUL"
+
+        self.uut.process_project_version(self.project, self.version)
+        assert_equals(shell_mock.mock_calls[1][1], ("gradle :printClasspath -b '/test dir/classpath.gradle'",))
+        assert_in(call("/path/dependency1.jar", self.dep_path), copy_mock.mock_calls)
+        assert_in(call("/path/dependency2.jar", self.dep_path), copy_mock.mock_calls)
+
+    @patch("tasks.implementations.compile.shutil.copy")
+    @patch("tasks.implementations.compile.Shell.exec")
+    def test_compile_with_gradle_project_dir_parameter_spaces2(self, shell_mock, copy_mock):
```

Comment:
The difference between the spaces/spaces2 tests is single vs. double quotes, right? Please change the test names to reflect that, e.g., test_compile_with_gradle_project_dir_parameter_space_single_quotes.